app/javascript/controllers/interest-places_controller.js

```diff
@@ -0,0 +1,31 @@
+import { Controller } from 'stimulus'; 
+export default class extends Controller {
+  static targets = ['zip_id', 'settlement_id']
+
+  connect() {
+    this.changeSettlements();
+  }
+
+  changeSettlements() {
+    let zip_id = this.zip_idTarget.value;
+    let current_value = this.settlement_idTarget.dataset.value;
+    // console.log(zip_id)
+    if (zip_id) {
+      fetch('/zips/'+zip_id+'.json').then(resp => resp.json())
+        .then(resp => {
+          // console.log(resp);
+          this.settlement_idTarget.innerHTML = '';
+          this.settlement_idTarget.appendChild(document.createElement("option"));
+          resp.settlements.forEach(sett => {
+            const option = document.createElement("option");
+            option.value = sett.id;
+            option.innerHTML = sett.name;
+            if (sett.id == current_value) {option.setAttribute('selected', 'selected')}
+            this.settlement_idTarget.appendChild(option);
+          });
+        }).catch(err => {
+          console.log(err);
+        });
+    }
+  }
```

Comment:
looks good, although you could use async/await to do this

**why should I do this with async/await?**

the thing is when you use promises you should use `then()` clause, which receives a callback function, the main problem of use callbacks is the famous [callback hell](http://callbackhell.com/). Nowadays Javascript has some awesome features to avoid that problem using [async/await](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await) syntax

```js
// this is a hint, you should not just copy and paste, probably need more code to achieve the same goal

async changeSettlements() {
    let zip_id = this.zip_idTarget.value;
    let current_value = this.settlement_idTarget.dataset.value;

    if (zip_id) {
      try {
        const response = await fetch(`/zips/${zip_id}.json`)
        const data = await response.json()

        this.settlement_idTarget.innerHTML = '';
        this.settlement_idTarget.appendChild(document.createElement("option"));
        data.settlements.forEach(({ id, name }) => {
          const option = document.createElement("option");
          option.value = id;
          option.innerHTML = name;
          if (id == current_value) {option.setAttribute('selected', 'selected')}
          this.settlement_idTarget.appendChild(option);
        });
      } catch (e) {
        console.error(e);
      }
    }
  }
```

Reply:
Thanks for the review, yes I noticed about the duplicate commits I dont know why that happened, It was after a rebase of master that had new commits in my working branch, I did not know how to solve it. 
About the async/await javascript I will check that, thanks for the recommendation and the hint, I did not have idea about that.

Reply:
I think you could drop the duplicated commits using an interactive rebase

`git rebase -i BASE_BRANCH`